packages/headless/src/api/commerce/common/product.ts

```diff
@@ -139,6 +139,10 @@ export interface BaseProduct {
    * The length and offset of each word to highlight in the product excerpt string.
    */
   excerptsHighlights?: HighlightKeyword[];
+  /**
+   * The ID of the response that returned the product.
+   */
+  responseId?: string;
```

Comment:
Why can this be optional? Search, Listing, and Recs calls all return a `responseId`